src/transformStateWithClones.js

```diff
@@ -7,7 +7,67 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const properties = {
+    addProperties: 'addProperties',
+    clear: 'clear',
+    removeProperties: 'removeProperties',
+  };
+  const newObj = { ...state };
+  let obj = {};
+  const newArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case properties.addProperties:
+        assignNewDataToState(newObj, action);
+        obj = writeToArray(obj, newObj, newArray);
+        break;
+
+      case properties.removeProperties:
+        for (const key of getRemoveKeys(action)) {
+          deleteFromState(newObj, key);
+        }
+        obj = writeToArray(obj, newObj, newArray);
```

Comment:
same here. no need to use `writeToArray`. you can always works with your `newObj` (it is copy of your `state` object)

Reply:
@anstsot if you write the changed newObj to the array, then the result is an array with 3 identical objects (the array is overwritten)